model/repo/src/main/java/io/sundr/model/repo/DefinitionRepository.java

```diff
@@ -179,18 +175,18 @@ public synchronized boolean hasDefinition(String fullyQualifiedName) {
     return definitions.containsKey(fullyQualifiedName) || suppliers.containsKey(fullyQualifiedName);
   }
 
+  public synchronized TypeDef getRealizedDefinition(String fullyQualifiedName) {
+    return definitions.get(fullyQualifiedName);
+  }
+
   public synchronized TypeDef getDefinition(String fullyQualifiedName) {
     if (definitions.containsKey(fullyQualifiedName)) {
       return definitions.get(fullyQualifiedName);
     }
     if (suppliers.containsKey(fullyQualifiedName)) {
       TypeDef typeDef = suppliers.get(fullyQualifiedName).get();
-      if (!definitions.containsKey(fullyQualifiedName)) {
-        definitions.put(fullyQualifiedName, typeDef);
-        return typeDef;
-      } else {
-        return definitions.get(fullyQualifiedName);
-      }
+      definitions.put(fullyQualifiedName, typeDef);
```

Comment:
Because of the synchronization it shouldn't make a difference.  Do you see a case where could be populated with alternative definition?  The supplier.get can cause further updates to the repository, but it shouldn't affect the current key - otherwise we're in a recursive scenario.